pkg/controllers/provisioning/provisioner.go

```diff
@@ -103,23 +107,26 @@ func (p *Provisioner) Trigger() {
 	p.batcher.Trigger()
 }
 
-func (p *Provisioner) Register(_ context.Context, mgr manager.Manager) error {
-	return controller.NewSingletonManagedBy(mgr).
+func (p *Provisioner) Register(_ context.Context, m manager.Manager) error {
+	return controllerruntime.NewControllerManagedBy(m).
 		Named("provisioner").
-		Complete(p)
+		WatchesRawSource(singleton.Source()).
+		Complete(singleton.AsReconciler(p))
 }
 
-func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (result reconcile.Result, err error) {
+func (p *Provisioner) Reconcile(ctx context.Context) (result reconcile.Result, err error) {
+	ctx = injection.WithControllerName(ctx, "provisioner")
+
 	// Batch pods
 	if triggered := p.batcher.Wait(ctx); !triggered {
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: controller.Immediately}, nil
```

Comment:
Double-checking, but this should be the same behavior as what we had before, right? Our singleton controller would requeue immediately on an empty result, whereas controller runtime will not.

Reply:
Correct.